xla/tests/onednn_layer_norm_test.cc

```diff
@@ -141,6 +141,75 @@ TEST_F(LayerNormTest, LayerNormTest0_F16) {
   MatchOptimizedHlo(layer_norm_module_str, onednn_layer_norm_);
 }
 
+// Test case encountered in models like TFViTForImageClassification in
+// HuggingFace
+// (https://huggingface.co/docs/transformers/model_doc/vit#transformers.TFViTForImageClassification)
+TEST_F(LayerNormTest, LayerNormTest1_BF16) {
+  const char* layer_norm_module_str = R"(
+  HloModule layer_norm.test, entry_computation_layout={(bf16[160,197,768]{2,1,0}, bf16[768]{0}, bf16[768]{0})->bf16[160,197,768]{2,1,0}}
+  region_add {
+    Arg_0.7555 = f32[] parameter(0)
+    Arg_1.7556 = f32[] parameter(1)
+    ROOT add.7557 = f32[] add(Arg_0.7555, Arg_1.7556)
+  }
+  ENTRY main {
+    Arg_0.1 = bf16[160,197,768]{2,1,0} parameter(0), sharding={replicated}
+    Arg_0.2 = bf16[768]{0} parameter(1), sharding={replicated}
+    Arg_0.3 = bf16[768]{0} parameter(2), sharding={replicated}
+    convert.80 = f32[160,197,768]{2,1,0} convert(Arg_0.1)
+    constant.81 = f32[] constant(0)
+    convert.82 = f32[] convert(constant.81)
+    reduce.87 = f32[160,197]{1,0} reduce(convert.80, convert.82), dimensions={2}, to_apply=region_add
+    constant.88 = s32[] constant(768)
+    convert.89 = f32[] convert(constant.88)
+    broadcast.90 = f32[160,197]{1,0} broadcast(convert.89), dimensions={}
+    divide.91 = f32[160,197]{1,0} divide(reduce.87, broadcast.90)
+    convert.92 = bf16[160,197]{1,0} convert(divide.91)
+    reshape.93 = bf16[160,197,1]{2,1,0} reshape(convert.92)
+    reshape.94 = bf16[160,197]{1,0} reshape(reshape.93)
```

Comment:
Why do we need `reshape.93` and `reshape.94`? If this is not part of the tricky pattern we need to recognize, let's simplify it.

Reply:
Same here, since we are trying to capture the exact HLO produced by keras.LayerNormalization, which generates 2 back to back reshapes. So yes, we are trying to match this particular pattern, which is also observed for other test cases in this file.